1.6/Base/Source/BigSmallFramework/DefPatches/RaceFuser/HumanlikeAnimalGenerator.cs

```diff
@@ -189,17 +194,17 @@ public static void GenerateAndRegisterHumanlikeAnimal(PawnKindDef aniPawnKind, T
                     bothComps.Add(comp);
                 }
             }
-            var filteredComps = bothComps
-                .Where(x =>
-                    compWhitelist.Contains(x.GetType().ToString(), StringComparer.OrdinalIgnoreCase) ||
-                    compWhitelist.Contains(x.compClass.ToString(), StringComparer.OrdinalIgnoreCase))
-                .ToList();
-
+			var filteredComps = bothComps
+				.Where(x =>
+					compWhitelist.Contains(x.GetType().ToString(), StringComparer.OrdinalIgnoreCase) ||
+					compWhitelist.Contains(x.compClass.ToString(), StringComparer.OrdinalIgnoreCase))
+				.Distinct()
+				.ToList();
 
+			newThing.comps = filteredComps;
+			newThing.thingClass = aniThing.thingClass;

```

Comment:
I'm not sure about this one. To allow dragons to fly in DD, Otters use a derived Pawn class, which would need to be reflected by the Sapient version.

However, i don't see any big issues with this currently, as most of the time, it should just be "Pawn"